app/src/main/java/com/framgia/music_49/screen/home/SongNewAndHotAdapter.java

```diff
@@ -0,0 +1,74 @@
+package com.framgia.music_49.screen.home;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.request.RequestOptions;
+import com.framgia.music_49.data.model.Song;
+import com.framgia_music_49.R;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SongNewAndHotAdapter extends RecyclerView.Adapter<SongNewAndHotAdapter.ItemHolder> {
+    private Context mContext;
+    private List<Song> mSongs;
+
+    public SongNewAndHotAdapter(Context context) {
+        mContext = context;
+        mSongs = new ArrayList<>();
+    }
+
+    public void updateData(List<Song> list) {
+        if (mSongs != null) {
+            mSongs.clear();
+        }
+        assert list != null;
+        mSongs.addAll(list);
+        notifyDataSetChanged();
+    }
+
+    @NonNull
+    @Override
+    public ItemHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
+        View view = LayoutInflater.from(viewGroup.getContext())
+                .inflate(R.layout.item_view_song_new_and_hot, viewGroup, false);
+        return new ItemHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ItemHolder viewHolder, int position) {
+        viewHolder.binData(mSongs.get(position));
+    }
+
+    @Override
+    public int getItemCount() {
+        return mSongs != null ? mSongs.size() : 0;
+    }
+
+    public class ItemHolder extends RecyclerView.ViewHolder {
```

Comment:
https://stackoverflow.com/questions/13516835/why-in-viewholder-pattern-should-the-viewholder-class-be-static